pydoctor/sphinx_ext/build_apidocs.py

```diff
@@ -38,26 +39,27 @@
 
 from pydoctor import __version__
 from pydoctor.driver import main, parse_args
-from pydoctor.sphinx_ext import get_source_reference
+from pydoctor.util import get_source_reference
 
 logger = logging.getLogger(__name__)
 
-# Shared state between init and finish.
-_placeholders: Dict[str, str] = {}
-
 
 def on_build_finished(app: Sphinx, exception: Exception) -> None:
     """
     Called when Sphinx build is done.
     """
+
+    # Share placeholders between init and finish.
+    placeholders = cast(Mapping[str, str], app.config['placeholders'])
```

Comment:
Since `__getitem__()` returns `Any`, you don't need a cast: you can assign to an annotated local variable instead:
```suggestion
    placeholders: Mapping[str, str] = app.config['placeholders']
```
This is a "softer" correction in the sense that `cast` overrides everything mypy knows about a type, while this annotation adds information without overriding what mypy already knows.

In this case it probably doesn't matter much, since I don't think `__getitem__()` will ever be able to return anything other than `Any`, but in cases where a value is `Any` because the library used doesn't have annotations yet, the variant without a cast is a bit safer.

Reply:
Thanks. Makes sense. I am still learning mypy. Many thanks for your support. Much appreciated :)

But if  try you suggestion I get 

```
pydoctor/sphinx_ext/build_apidocs.py:55: error: Incompatible types in
assignment (expression has type "str", variable has type "Mapping[str, str]") 
[assignment]
        placeholders: Mapping[str, str] = app.config['placeholders']
                                          ^
Found 1 error in 1 file (checked 43 source files)
```

Reply:
Hmm, for some reason `sphinx.config.Config.__getitem__()` is annotated to return `str`, while `__setitem__()` accepts `Any`. I don't know if this is intentional or a mistake in their annotations.

You can work around it by using `__getattr__()` instead, which does return `Any`:
```py
    placeholders: Mapping[str, str] = app.config.placeholders
```